packages/forms/backend/sirius-components-collaborative-widget-reference/src/main/java/org/eclipse/sirius/components/collaborative/widget/reference/handlers/ClearReferenceEventHandler.java

```diff
@@ -0,0 +1,101 @@
+import org.eclipse.sirius.components.collaborative.widget.reference.dto.ClearReferenceInput;
+import org.eclipse.sirius.components.core.api.ErrorPayload;
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.core.api.IPayload;
+import org.eclipse.sirius.components.core.api.SuccessPayload;
+import org.eclipse.sirius.components.forms.Form;
+import org.eclipse.sirius.components.representations.Failure;
+import org.eclipse.sirius.components.representations.IStatus;
+import org.eclipse.sirius.components.representations.Success;
+import org.eclipse.sirius.components.widget.reference.ReferenceWidget;
+import org.springframework.stereotype.Service;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import reactor.core.publisher.Sinks.Many;
+import reactor.core.publisher.Sinks.One;
+
+/**
+ * The handler to handle the clear of the reference value event.
+ *
+ * @author Jerome Gout
+ */
+@Service
+public class ClearReferenceEventHandler implements IFormEventHandler {
+
+    private final ICollaborativeFormMessageService messageService;
+
+    private final Counter counter;
+
+    private final IFormQueryService formQueryService;
+
+    public ClearReferenceEventHandler(IFormQueryService formQueryService, ICollaborativeFormMessageService messageService, MeterRegistry meterRegistry) {
+        this.formQueryService = Objects.requireNonNull(formQueryService);
+        this.messageService = Objects.requireNonNull(messageService);
+
+        this.counter = Counter.builder(Monitoring.EVENT_HANDLER)
+                .tag(Monitoring.NAME, this.getClass().getSimpleName())
+                .register(meterRegistry);
+    }
+
+    @Override
+    public boolean canHandle(IFormInput formInput) {
+        return formInput instanceof ClearReferenceInput;
+    }
+
+    @Override
+    public void handle(One<IPayload> payloadSink, Many<ChangeDescription> changeDescriptionSink, IEditingContext editingContext, Form form, IFormInput formInput) {
+        this.counter.increment();
+
+        String message = this.messageService.invalidInput(formInput.getClass().getSimpleName(), ClearReferenceInput.class.getSimpleName());
+        IPayload payload = new ErrorPayload(formInput.id(), message);
+        ChangeDescription changeDescription = new ChangeDescription(ChangeKind.NOTHING, formInput.representationId(), formInput);
+
+        if (formInput instanceof ClearReferenceInput input) {
+
+            var optionalReferenceWidget = this.formQueryService.findWidget(form, input.referenceWidgetId())
+                    .filter(ReferenceWidget.class::isInstance)
+                    .map(ReferenceWidget.class::cast);
+
+            IStatus status;
+            if (optionalReferenceWidget.map(ReferenceWidget::isReadOnly).filter(Boolean::booleanValue).isPresent()) {
+                status = new Failure("Read-only widget can not be edited");
```

Comment:
Why are messages not internationalized? (And that's not the only one, far from it)

Reply:
I didn't know that there was a mechanism in place to internationalize messages.
Please, point out some examples to follow.

Reply:
You'll find an example of the pattern used elsewhere for this in the pacakge `org.eclipse.sirius.components.collaborative.diagrams.messages`. The actual messages are externalized in a `properties` file, here `/sirius-components-collaborative-diagrams/src/main/resources/messages/sirius-web-spring-collaborative-diagrams.properties`